core/runtime/impl/core_impl.cpp

```diff
@@ -0,0 +1,100 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "runtime/impl/core_impl.hpp"
+
+#include "primitives/scale_codec.hpp"
+#include "runtime/impl/runtime_external_interface.hpp"
+#include "runtime/impl/wasm_memory_impl.hpp"
+#include "runtime/impl/wasm_memory_stream.hpp"
+
+namespace kagome::runtime {
+
+  CoreImpl::CoreImpl(common::Buffer state_code,
+                     std::shared_ptr<extensions::Extension> extension,
+                     std::shared_ptr<primitives::ScaleCodec> codec)
+      : state_code_(std::move(state_code)),
+        memory_(extension->memory()),
+        executor_(std::move(extension)),
+        codec_(std::move(codec)) {}
+
+  outcome::result<primitives::Version> CoreImpl::version() {
+    uint64_t version_long =
+        executor_
+            .call(state_code_, "Core_version",
+                  wasm::LiteralList({wasm::Literal(0), wasm::Literal(0)}))
+            .geti64();
+
+    // first 32 bits are address and second are length (length not needed)
+    runtime::WasmPointer version_address = version_long & 0xFFFFFFFFLLU;
+
+    WasmMemoryStream stream(memory_);
+
+    OUTCOME_TRY(stream.advance(version_address));
+
+    OUTCOME_TRY(version, codec_->decodeVersion(stream));
+
+    return std::move(version);
+  }
+
+  outcome::result<void> CoreImpl::execute_block(
+      const kagome::primitives::Block &block) {
+    OUTCOME_TRY(encoded_block, codec_->encodeBlock(block));
+
+    runtime::SizeType block_size = encoded_block.size();
+    runtime::WasmPointer ptr = memory_->allocate(block_size);
+    memory_->storeBuffer(ptr, encoded_block);
+
+    executor_
+        .call(
+            state_code_, "Core_execute_block",
+            wasm::LiteralList({wasm::Literal(ptr), wasm::Literal(block_size)}))
+        .geti64();
+
+    return outcome::success();
+  }
+
+  outcome::result<void> CoreImpl::initialise_block(
+      const kagome::primitives::BlockHeader &header) {
+    OUTCOME_TRY(encoded_header, codec_->encodeBlockHeader(header));
+
+    runtime::SizeType header_size = encoded_header.size();
+    runtime::WasmPointer ptr = memory_->allocate(header_size);
+    memory_->storeBuffer(ptr, encoded_header);
+
+    executor_
+        .call(
+            state_code_, "Core_initialise_block",
+            wasm::LiteralList({wasm::Literal(ptr), wasm::Literal(header_size)}))
+        .geti64();
```

Comment:
Why is the result of the call obtained here with geti64 but not used afterwards?

Reply:
`(func $Core_execute_block (type 23) (param i32 i32) (result i64)`
This is the declaration of this function in runtime. If we get other type than i64, then code fails, as there is an assertion inside checking that returned value matches the type  of the literal